改完的（改正第一个方法）

```diff
@@ -0,0 +1,67 @@
+public class Phrase
+{
+    private static String currentPhrase;
+
+    /** Constructs a new Phrase object. */
+    public Phrase(String p)
+    {
+         currentPhrase = p;
+    }
+
+    /** Returns the index of the nth occurrence of str in the current phrase;
+     *  returns -1 if the nth occurrence does not exist.
+     *  Precondition: str.length() > 0 and n > 0
+     *  Postcondition: the current phrase is not modified.
+     */
+    public static int findNthOccurrence(String str, int n)
+    {
+        /* implementation not shown */
+        int loc = currentPhrase.indexOf(str);
+        int i=1;
+        int a=currentPhrase.subString(loc+1)).indexOf(str);
+        while(i<=n && a!=-1){
+        loc=(currentPhrase.subString(loc+1)).indexOf(str))+loc+1;
+        i++;
+        a=currentPhrase.subString(loc+1)).indexOf(str);
+        }
+        return loc;
+    }
```

Comment:
Consider a case: you want to find 5th occurrence, but there just 3 occurrence of str in currentPhrase, what will be return?